Relax assertions when both PER CALL values are equal (Erlang/OTP 28)

Eiji7 opened the pull request:
Fixes: #14564

This fix implements `Erlang/OTP` version check. The alternative solution could be:
```elixir
# extract results
[_warmup, _profile_results, _columns, _total, first, second, _profile_done, ""] =
  String.split(result, ~r/\n+/)

# split columns
list = Enum.map([first, second], &String.split(&1, ~r/\s+/))

sort_func = fn result_colls ->
  result_colls
  |> List.last()
  |> String.to_float()
end

list == Enum.sort_by(list, sort_func)
```

The alternative solution additionally does not relies on `3.\d{2}` value pattern of `PER CALL` column.

josevalim commented:
I like your alternative solution, relying less on regexes, but we should also check on the calls. Perhaps we build `{per_call, mfa}` tuples and assert on the names after sorting?

Eiji7 commented:
Oh, wait - I have just realised something something … This is the **only** test that uses `--sort` argument with `per_call` value. We can just sort by `calls` and then verify the sorted result. For some reason my mind was fixed on not changing this task call at all. :sweat_smile:

Eiji7 commented:
@josevalim I have reverted back to old regex solution, because it's shorter, but I kept the change in `--sort` argument value, so now we are sure that they are always properly sorted regardless of `JIT`, architecture or `Erlang/OTP` version. This resulted in swap the order of expected lines, so now `Enum.each/2` is expected first. I have also added patterns to verify the `calls` column value (1 and 5 surrounded by whitespaces and right after mfa).
